etna/datasets/tsdataset.py

```diff
@@ -871,50 +871,72 @@ def train_test_split(
 
         return train, test
 
-    def update_columns_from_pandas(self, df: pd.DataFrame, regressors: Optional[List[str]] = None):
-        """Update the dataset with the new columns from pandas dataframe.
+    def update_columns_from_pandas(self, df_update: pd.DataFrame):
+        """Update the existing columns in the dataset with the new values from pandas dataframe.
 
-        It is recommended to add the exogenous regressor using the constructor of the TSDataset.
-        This method does not add the regressors as exogenous data in df_exog, but only update the df attribute.
+        Before updating columns in df, columns of df_update will be cropped by the last timestamp in df.
+        If columns in df_exog are not updated. If you wish to update the df_exog, create the new
+        instance of TSDataset.
 
         Parameters
         ----------
-        df:
-            Dataframe with the new columns in wide ETNA format.
-            If columns with the same names already exist in the dataset, then values will be updated.
-        regressors:
-            List of regressors in the passed dataframe
+        df_update:
+            Dataframe with new values in wide ETNA format.
         """
-        columns_in_dataset = self.columns.get_level_values("feature")
-        new_columns = df.columns.get_level_values("feature")
-
-        columns_to_update = list(set(columns_in_dataset) & set(new_columns))
-        self.df.loc[:, self.idx[self.segments, columns_to_update]] = df.loc[
-            :, self.idx[self.segments, columns_to_update]
+        columns_to_update = sorted(set(df_update.columns.get_level_values("feature")))
+        self.df.loc[:, self.idx[self.segments, columns_to_update]] = df_update.loc[
```

Comment:
What about problem with selection multiple columns?

Reply:
Add link in the corresponding issue